### PR TITLE
ENT-3551: Add new health diagnostic two conditions.

### PR DESCRIPTION
**Pull request by aleksandrychev**

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@craigcomstock please review and merge. thanks!

File: enterprise-cfengine-guide/hosts-health.markdown [hunk 1]
```diff
@@ -21,11 +21,13 @@ Here you will find extensive information on single hosts that CFEngine detects a
...
 
-![Hosts](Mission-Portal-Hosts-1.png)
+![Hosts](Mission-portal-health-dignostics-header.png)
 
 You can get quick access to the health of hosts, including direct links to reports, from the Health drop down at the top of every Enterprise UI screen. Hosts are listed as unhealthy if:
 
 * the hub was not able to connect to and collect data from the host within a set time interval (unreachable host). The time interval can be set in the Mission Portal settings.
 * the policy did not get executed for the last three runs. This could be caused by `cf-execd` not running on the host (scheduling deviation) or an error in policy that stops its execution. The hub is still able to contact the host, but it will return stale data because of this deviation.
+* two or more hosts use the same key.This is detected by the IP address tied to a CFEngine key has changed the last 3 scheduled runs (by default, configurable in settings) in a row.
+* reports have recently been collected, but cf-agent has not recently “gracefully” run. “Recently” is defined by the configured run-interval of their cf-agent.
```
Comment (craigcomstock) [inline enterprise-cfengine-guide/hosts-health.markdown]: I would reword the first addition to something like: "two or more hosts use the same key. This is detected if the IP address tied to a CFEngine key has changed in the last three scheduled runs. The number of scheduled runs that cause the unhealthy status is configurable in settings." But also add WHERE the settings are and what variable name has that setting.

Comment (craigcomstock) [reply enterprise-cfengine-guide/hosts-health.markdown]: What is the definition of "gracefully"?

Comment (roschern) [reply enterprise-cfengine-guide/hosts-health.markdown]: Lets just remove the word gracefully. It doesn't mean anything in this context. Either it runs and reports, or it doesn't. Not gracefully basically means it crashed, or otherwise failed in some way.

Comment (aleksandrychev) [reply enterprise-cfengine-guide/hosts-health.markdown]: >But also add WHERE the settings are and what variable name has that setting.

>Lets just remove the word gracefully.

OK, I will do

Comment (aleksandrychev) [reply enterprise-cfengine-guide/hosts-health.markdown]: Done

File: enterprise-cfengine-guide/hosts-health.markdown [hunk 2]
```diff
@@ -21,11 +21,14 @@ Here you will find extensive information on single hosts that CFEngine detects a
...
 
-![Hosts](Mission-Portal-Hosts-1.png)
+![Hosts](Mission-portal-health-dignostics-header.png)
 
 You can get quick access to the health of hosts, including direct links to reports, from the Health drop down at the top of every Enterprise UI screen. Hosts are listed as unhealthy if:
 
 * the hub was not able to connect to and collect data from the host within a set time interval (unreachable host). The time interval can be set in the Mission Portal settings.
 * the policy did not get executed for the last three runs. This could be caused by `cf-execd` not running on the host (scheduling deviation) or an error in policy that stops its execution. The hub is still able to contact the host, but it will return stale data because of this deviation.
+* two or more hosts use the same key. This is detected if the IP address tied to a CFEngine key has changed in the last three scheduled runs. The number of scheduled runs that cause the unhealthy status is configurable in [settings][Settings#Preferences].
+* reports have recently been collected, but cf-agent has not recently “gracefully” run. “Recently” is defined by the configured run-interval of their cf-agent.
```
Comment (craigcomstock) [inline enterprise-cfengine-guide/hosts-health.markdown]: the setting is named "Number of samples used to identify a duplicate identity" right?

Comment (craigcomstock) [reply enterprise-cfengine-guide/hosts-health.markdown]: Didn't @roschern want "gracefully" removed or replaced here? Maybe a comment in e-mail? I can't find it now.

Comment (aleksandrychev) [reply enterprise-cfengine-guide/hosts-health.markdown]: @craigcomstock  Ah I've forgotten about gracefully. One moment

>the setting is named "Number of samples used to identify a duplicate identity" right? 

right

Comment (aleksandrychev) [reply enterprise-cfengine-guide/hosts-health.markdown]: done
